### PR TITLE
community: update package name to bagelML

### PR DESCRIPTION
**Pull request by ihpolash**
**Description**
This pull request updates the Bagel Network package name from "betabageldb" to "bagelML" to align with the latest changes made by the Bagel Network team.

The following modifications have been made:

- Updated all references to the old package name ("betabageldb") with the new package name ("bagelML") throughout the codebase.
- Modified the documentation, and any relevant scripts to reflect the package name change.
- Tested the changes to ensure that the functionality remains intact and no breaking changes were introduced.

By merging this pull request, our project will stay up to date with the latest Bagel Network package naming convention, ensuring compatibility and smooth integration with their updated library.

Please review the changes and provide any feedback or suggestions. Thank you!

### CODE REVIEW

**Comment by ihpolash** [comment]
@efriis @baskaryan could you please review the PR. We are rebranding Bagel from BagelDB. If you have any comments, feel free to ping me. Thank you.

**Comment by ihpolash** [comment]
@liugddx could you please review Bagel's Latest Update?

**Comment by Rijul1204** [comment]
@ccurme, Could you please review this PR? We have been trying to get it reviewed for a very long time. Any help would be much appreciated. Thanks.

**Comment by ihpolash** [comment]
@efriis please could you make the PR review approval? we have been looking forward to this.

File: libs/langchain/langchain/vectorstores/bagel.py
Comment (baskaryan) [inline libs/langchain/langchain/vectorstores/bagel.py:1]: can we remove all changes to `libs/langchain`? community imports here are for backwards compatibility only and will be removed in 0.2 soon

File: libs/community/langchain_community/vectorstores/bagel.py
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/bagel.py:1]: for backwards compatibility we should keep around a `bageldb.py` file that imports and exposes Bagel

```python
from langchain_communiuty.vectorstores.bagel import Bagel

__all__ = ["Bagel"]
```

Comment (ihpolash) [reply libs/community/langchain_community/vectorstores/bagel.py:1]: @baskaryan Thank you for the review. I have made changes for backwards compatibility. could you please review again?

File: libs/langchain/langchain/vectorstores/bagel.py
Comment (baskaryan) [reply libs/langchain/langchain/vectorstores/bagel.py:1]: bump on this, shouldn't be any changes in `libs/langchain` at all. we're currently working to remove langchain_community imports from langchain altogether

Comment (ihpolash) [reply libs/langchain/langchain/vectorstores/bagel.py:1]: @baskaryan I have fixed the issues. could you make reviews again?

File: libs/community/langchain_community/vectorstores/bageldb.py
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/bageldb.py:1]: this file can jsut import from bagel.py

```python
from langchain_community.vectorstores.bagel import Bagel

__all__ = ["Bagel"]
```

**Review by baskaryan — COMMENTED** [review]
happy to help get this over the finish line if you want to give me edit access to the fork

File: libs/community/langchain_community/vectorstores/bageldb.py
Comment (ihpolash) [reply libs/community/langchain_community/vectorstores/bageldb.py:1]: @baskaryan I have addressed your reviews. Also, we have provided you temporary edit access to the fork. Thank you.

**Review by baskaryan — APPROVED** [review]

**Comment by ihpolash** [comment]
@baskaryan could you please make review & make the merge? Thank you.

**Pull request merged**